circumstellar-habitable-zone-simulator/src/diagram.jsx

```diff
@@ -105,16 +116,29 @@ export default class CSHZDiagram extends React.Component {
         // Returns the planet's position in pixels in the PIXI scene and the pixels
         // per AU to draw the scene
 
-        // First determine a zoom level
-        let zoomLevel = 0;
-        let pixelsFromStar = this.auToPixels(planetDistance, this.zoomLevels[zoomLevel].pixelsPerAU);
+        // First determine a value for pixels per AU
+        let pixelsPerAU = this.zoomLevels[0].pixelsPerAU;
+        let pixelsFromStar = this.auToPixels(planetDistance, pixelsPerAU);
         let planetXPosition = STAR_ORIGIN_POINT[0] + pixelsFromStar;
-        while (planetXPosition > ZOOM_UPPER_BREAKPOINT) {
-             zoomLevel++;
-             pixelsFromStar = this.auToPixels(this.props.planetDistance, this.zoomLevels[zoomLevel].pixelsPerAU);
-             planetXPosition = STAR_ORIGIN_POINT[0] + pixelsFromStar;
+        const lastZoomIdx = this.zoomLevels.length - 1;
+        while (planetXPosition > ZOOM_UPPER_BREAKPOINT && pixelsPerAU >= this.zoomLevels[lastZoomIdx].pixelsPerAU) {
+            if (pixelsPerAU < this.zoomLevels[lastZoomIdx].pixelsPerAU) {
```

Comment:
I think the loop invariant `pixelsPerAU >= this.zoomLevels[lastZoomIdx].pixelsPerAU` eliminates the need for this if statement? But maybe I'm missing something?

Reply:
Nope didn't miss anything, just me being neurotic.